api-get-object.go

```diff
@@ -148,65 +234,107 @@ func (o *Object) Read(b []byte) (n int, err error) {
 	o.mutex.Lock()
 	defer o.mutex.Unlock()
 
-	// prevErr is previous error saved from previous operation.
-	if o.prevErr != nil || o.isClosed {
-		return 0, o.prevErr
-	}
+	// This is the first request.
+	if !o.isStarted {
+		o.isStarted = true
```

Comment:
Set isStarted as true only after we have successfully completed a channel handshake.